analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.cs

```diff
@@ -290,6 +290,21 @@ public void ToString_WithAll()
 ");
         }
 
+        [TestMethod]
+        public void ProgramStateToArrayEncapsulatesSingleElement()
+        {
+            var programState = ProgramState.Empty;
+            ProgramState[] target = programState;
+            target.Should().NotBeNull().And.HaveCount(1).And.Equal(new[] { ProgramState.Empty });
+        }
+
+        [TestMethod]
+        public void ProgramStateToArrayConversionHandlesNull()
+        {
+            ProgramState[] target = (ProgramState)null;
+            target.Should().NotBeNull().And.HaveCount(0).And.BeOfType<ProgramState[]>();
```

Comment:
The `BeOfType` doesn't make sense on explicit `ProgramState[]` variable declaration anyway
```suggestion
            target.Should().BeEmpty();
```